src/fparser/two/tests/fortran2003/test_type_decl_stmt_r501.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) 2022 Science and Technology Facilities Council.
```

Comment:
Thanks for moving the tests into their own file. The only problem I have with this is that the tests have not been updated (apart from the symbol table part), they have been simply copied and I suspect they will need to be improved/extended e.g. what about `match2`. Could you add a new issue to do this and reference it here. You could also reference #259 from here too.

Reply:
Done.